libs/vkd3d/resource.c

```diff
@@ -3513,6 +3513,88 @@ void d3d12_desc_copy(struct d3d12_desc *dst, struct d3d12_desc *src,
     }
 }
 
+void d3d12_desc_copy_mutable(struct d3d12_desc *dst, struct d3d12_desc *src,
+        unsigned int count, D3D12_DESCRIPTOR_HEAP_TYPE heap_type, struct d3d12_device *device)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+    VkCopyDescriptorSet vk_copies[2], *vk_copy;
+    struct vkd3d_descriptor_binding binding;
+    enum vkd3d_bindless_set_flag flag;
+    uint32_t copy_count = 0;
+    unsigned int i;
+
+    flag = heap_type == D3D12_DESCRIPTOR_HEAP_TYPE_SAMPLER ? VKD3D_BINDLESS_SET_SAMPLER : VKD3D_BINDLESS_SET_MUTABLE;
+    binding = vkd3d_bindless_state_find_set(&device->bindless_state, flag);
```

Comment:
Is it possible to cache the binding for these sets? It feels a little suboptimal to scan through a binding list since this is in the ultra-hot path.

Reply:
We're only going to have two bindings in total in this case though, and this path shouldn't be quite as hot if the app copies consecutive ranges.

We could in theory store this with the descriptor heap, but it's **only** going to work with the mutable type extension, and completely meaningless otherwise.